crates/proto/src/xfer/dnssec_dns_handle.rs

```diff
@@ -448,7 +448,31 @@ where
             continue;
         };
 
+        let Ok(key_tag) = key_rdata.calculate_key_tag() else {
+            continue;
+        };
+
         for r in ds_records.iter() {
+            if r.data().algorithm() != key_rdata.algorithm() {
+                debug!(
```

Comment:
Suggestion: make this and the other call at line 467 trace! to keep the noise down in the log file (these are going to fire on pretty much every query for a DNSSEC zone), and also nuke the trace! call at line 489; it is superfluous given the debug! call at line 480.